collections/src/main/scala/strawman/collection/ArrayOps.scala

```diff
@@ -54,11 +69,158 @@ class ArrayOps[A](val xs: Array[A])
+    override def equals(that: Any) = that match {
+      case that: ofShort => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofChar(val array: Array[Char]) extends ArrayView[Char] with Serializable {
+    def elemTag = ClassTag.Char
+    def length: Int = array.length
+    def apply(index: Int): Char = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofChar => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofInt(val array: Array[Int]) extends ArrayView[Int] with Serializable {
+    def elemTag = ClassTag.Int
+    def length: Int = array.length
+    def apply(index: Int): Int = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofInt => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofLong(val array: Array[Long]) extends ArrayView[Long] with Serializable {
+    def elemTag = ClassTag.Long
+    def length: Int = array.length
+    def apply(index: Int): Long = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofLong => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofFloat(val array: Array[Float]) extends ArrayView[Float] with Serializable {
+    def elemTag = ClassTag.Float
+    def length: Int = array.length
+    def apply(index: Int): Float = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofFloat => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofDouble(val array: Array[Double]) extends ArrayView[Double] with Serializable {
+    def elemTag = ClassTag.Double
+    def length: Int = array.length
+    def apply(index: Int): Double = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofDouble => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofBoolean(val array: Array[Boolean]) extends ArrayView[Boolean] with Serializable {
+    def elemTag = ClassTag.Boolean
+    def length: Int = array.length
+    def apply(index: Int): Boolean = array(index)
+    override def hashCode = arrayViewHash(array)
+    override def equals(that: Any) = that match {
+      case that: ofBoolean => Arrays.equals(array, that.array)
+      case _ => super.equals(that)
+    }
+  }
+
+  final class ofUnit(val array: Array[Unit]) extends ArrayView[Unit] with Serializable {
+    def elemTag = ClassTag.Unit
+    def length: Int = array.length
+    def apply(index: Int): Unit = array(index)
+    override def hashCode = arrayViewHash(array)
```

Comment:
Why do we want to override `ArrayView`’s hashcode?

Reply:
To make use of specialization